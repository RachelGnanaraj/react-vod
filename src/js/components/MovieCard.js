import React from 'react';

import MovieCardDescription from './MovieCardDescription'

import './MovieCard.scss'

export default class MovieCard extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      hover: false,
      playing: false,
    };
  }

  onMouseOver = () => {
    this.setState({hover: true}, () => {
      this.getVideo().play()
    })
  }

  onMouseLeave = () => {
    this.setState({hover: false}, () => {
      this.getVideo().pause()
    })
  }

  getVideo = () => this.video

  getClassName = () => {
    if (this.state.hover) return 'thumbnail movie-card movie-card--hover'

    return 'thumbnail movie-card movie-card--not-hover'
  }

  render() {
    const { id, url, title, description } = this.props
    const { hover } = this.state
    return (
      <div
        className={this.getClassName()}
        key={id.toString()}
        onMouseOver={this.onMouseOver}
        onMouseLeave={this.onMouseLeave}>
        <video ref={v => (this.video = v)}>
          <source src={url} type="video/mp4" />
        </video>
        <div className="caption">
          <h5>{title}</h5>
          <MovieCardDescription description={description} expanded={hover}/>
        </div>
      </div>
    )
  }
}
