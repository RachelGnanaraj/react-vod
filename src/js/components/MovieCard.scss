.movie-card {
    margin: 0 10px;
    position: 'relative';
    video {
        width: 100%    !important;
        height: auto   !important;
    }
}

.movie-card--not-hover h5{
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
}

.movie-card--not-hover .movie-description{
    overflow: hidden;
    text-overflow: ellipsis;
    max-height: 80px;
    line-height: 20px;
}

.movie-card .movie-description{
    min-height: 80px;
}