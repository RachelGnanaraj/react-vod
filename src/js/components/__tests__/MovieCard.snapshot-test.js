import React from 'react';
import MovieCard from '../MovieCard';
import renderer from 'react-test-renderer';

test('MovieCard changes when hovered', () => {
  const component = renderer.create(
    <MovieCard title='Test' description='test' url='test' id='1'/>,
  );
  const instance = component.getInstance()
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  instance.getVideo = () => ({
    play: () => {}
  })
  
  tree.props.onMouseOver();

  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});