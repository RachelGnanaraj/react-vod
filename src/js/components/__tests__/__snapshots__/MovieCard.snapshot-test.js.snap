// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard changes when hovered 1`] = `
<div
  className="thumbnail movie-card movie-card--not-hover"
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
>
  <video>
    <source
      src="test"
      type="video/mp4"
    />
  </video>
  <div
    className="caption"
  >
    <h5>
      Test
    </h5>
    <p
      className="card-text movie-description"
    >
      test
    </p>
  </div>
</div>
`;

exports[`MovieCard changes when hovered 2`] = `
<div
  className="thumbnail movie-card movie-card--hover"
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
>
  <video>
    <source
      src="test"
      type="video/mp4"
    />
  </video>
  <div
    className="caption"
  >
    <h5>
      Test
    </h5>
    <p
      className="card-text movie-description"
    >
      test
    </p>
  </div>
</div>
`;
